agent/pkg/proto/agent.proto

```diff
@@ -143,6 +143,7 @@ message RefreshResult {
   string stderr = 2;
   UpdateSummary summary = 3;
   optional string permalink = 4;
+  optional bool is_stack_locked = 5;
```

Comment:
I would definitely prefer an error structure of some kind, and probably nor embedded into the result struct.  Maybe the method can return a oneof?